### PR TITLE
Fix #562

### LINKED ISSUE #562: WGDashboard constantly throws you to the authorization page
**Describe The Problem**
Hello, after a while, the WGDashboard constantly throws you to the authorization page. If you click "back (arrow)" on the authorization page in the browser, it continues to work, but after a while it throws you back to the authorization page.
How to fix this? Please help.

**Expected Error / Traceback**
```
[2024-12-26 15:47:10 +0300] [77210] [INFO] Starting gunicorn 23.0.0
[2024-12-26 15:47:10 +0300] [77210] [INFO] Listening at: http://192.168.220.160:10086 (77210)
[2024-12-26 15:47:10 +0300] [77210] [INFO] Using worker: gthread
[2024-12-26 15:47:10 +0300] [77212] [INFO] Booting worker with pid: 77212
[WGDashboard] Background Thread #1 Started
[WGDashboard] Background Thread #2 Started
[2024-12-26 15:48:15,198] ERROR in app: Exception on /api/getWireguardConfigurationInfo [GET]
Traceback (most recent call last):
  File "/srv/WGDashboard/src/venv/lib/python3.11/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/venv/lib/python3.11/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/venv/lib/python3.11/site-packages/flask_cors/extension.py", line 194, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/venv/lib/python3.11/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/venv/lib/python3.11/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/dashboard.py", line 2253, in API_getConfigurationInfo
    return ResponseObject(data={
           ^^^^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/dashboard.py", line 54, in ResponseObject
    response = Flask.make_response(app, {
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/venv/lib/python3.11/site-packages/flask/app.py", line 1231, in make_response
    rv = self.json.response(rv)
         ^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/venv/lib/python3.11/site-packages/flask/json/provider.py", line 214, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/venv/lib/python3.11/site-packages/flask/json/provider.py", line 179, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/usr/lib/python3.11/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/dashboard.py", line 74, in default
    return o.toJson()
           ^^^^^^^^^^
  File "/srv/WGDashboard/src/dashboard.py", line 1184, in toJson
    self.getShareLink()
  File "/srv/WGDashboard/src/dashboard.py", line 1286, in getShareLink
    self.ShareLink = AllPeerShareLinks.getLink(self.configuration.Name, self.id)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/dashboard.py", line 410, in getLink
    self.__getSharedLinks()
  File "/srv/WGDashboard/src/dashboard.py", line 405, in __getSharedLinks
    allLinks = sqlSelect("SELECT * FROM PeerShareLinks WHERE ExpireDate IS NULL OR ExpireDate > datetime('now', 'localtime')").fetchall()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/WGDashboard/src/dashboard.py", line 1628, in sqlSelect
    with sqldb:
SystemError: error return without exception set
```

Raspberry Pi 3
**OS Information:**
 - OS: Raspbian12 PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
 - Python Version: Python 3.12.5

**Sample of your `.conf` file**
```
[Interface]
Address = 33.33.33.1/24
DNS = 127.0.0.1
PostUp = iptables -I INPUT -p udp --dport 34728 -j ACCEPT
PostUp = iptables -I FORWARD -i eth0 -o wg0 -j ACCEPT
PostUp = iptables -I FORWARD -i wg0 -j ACCEPT
PostUp = iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
PostUp = ip rule add from `ip addr show $(ip route | awk '/default/ { print $5 }') | grep "inet" | grep -v "inet6" | head -n 1 | awk '/inet/ {print $2}' | awk -F/ '{print $1}'` table main
PostDown = iptables -D INPUT -p udp --dport 34728 -j ACCEPT
PostDown = iptables -D FORWARD -i eth0 -o wg0 -j ACCEPT
PostDown = iptables -D FORWARD -i wg0 -j ACCEPT
PostDown = iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE
PostDown = ip rule del from `ip addr show $(ip route | awk '/default/ { print $5 }') | grep "inet" | grep -v "inet6" | head -n 1 | awk '/inet/ {print $2}' | awk -F/ '{print $1}'` table main
ListenPort = 34728
FwMark = 0xca6c
PrivateKey = ABCD...

[Peer]
PublicKey = EuKyomV9f2nK/FPXt+5OUla3IKON6Nn/3WtYPv1zcVo=
AllowedIPs = 33.33.33.111/32, 0.0.0.0/0
Endpoint = x.x.x.x:53628

[Peer]
PublicKey = M+bGYfx6NNhRnTyJgLZ1NTaS5Mzr1hE4b07cy1exxF0=
AllowedIPs = 33.33.33.2/32, 192.168.5.0/24
Endpoint = x.x.x.x:44582
```

### PR DESCRIPTION
**Pull request by donaldzou**
- Closing issue #562 
- It might due to some mysterious system error

### CODE REVIEW

File: src/dashboard.py [hunk 1]
```diff
@@ -1629,10 +1629,12 @@ def sqlSelect(statement: str, paramters: tuple = ()) -> sqlite3.Cursor:
         try:
             cursor = sqldb.cursor()
             return cursor.execute(statement, paramters)
-
         except sqlite3.OperationalError as error:
             print("[WGDashboard] SQLite Error:" + str(error) + " | Statement: " + statement)
             return []
+        except Exception as e:
```
Comment (DaanSelen) [inline src/dashboard.py:1635]: Is this not the same as the line above?

File: src/dashboard.py [hunk 2]
```diff
@@ -1645,6 +1647,9 @@ def sqlUpdate(statement: str, paramters: tuple = ()) -> sqlite3.Cursor:
             sqldb.commit()
         except sqlite3.OperationalError as error:
             print("[WGDashboard] SQLite Error:" + str(error) + " | Statement: " + statement)
+        except Exception as e:
+            print("[WGDashboard] SQLite Error:" + str(error) + " | Statement: " + statement)
```
Comment (DaanSelen) [inline src/dashboard.py:1651]: Can't this also be caught with a general exception instead of Exception AND sqlite3.operationalError?

Comment (donaldzou) [reply src/dashboard.py:1651]: Yes you're right haha.. Post-Christmas break syndrome ;)

File: src/dashboard.py (hunk 1 above)
Comment (donaldzou) [reply src/dashboard.py:1635]: Yes thats right

File: src/dashboard.py (hunk 2 above)
Comment (DaanSelen) [reply src/dashboard.py:1651]: No worries! That's why the review is handy!

File: src/dashboard.py (hunk 1 above)
Comment (DaanSelen) [reply src/dashboard.py:1635]: Let's keep it lean and efficient!

**Comment by donaldzou** [comment]
@DaanSelen should be good now :)

File: src/dashboard.py [hunk 3]
```diff
@@ -116,7 +116,6 @@ def __createLogDatabase(self):
                 self.loggerdb.commit()
     
     def log(self, URL: str = "", IP: str = "", Status: str = "true", Message: str = "") -> bool:
-        pass
```
Comment (DaanSelen) [inline src/dashboard.py:119]: What is the pass then try?

File: src/dashboard.py [hunk 4]
```diff
@@ -289,7 +288,7 @@ def saveJob(self, Job: PeerJob) -> tuple[bool, list] | tuple[bool, str]:
         except Exception as e:
             return False, str(e)
 
-    def deleteJob(self, Job: PeerJob) -> tuple[bool, list] | tuple[bool, str]:
+    def deleteJob(self, Job: PeerJob, deletedFrom = 'Job Runner') -> tuple[bool, list] | tuple[bool, str]:
```
Comment (DaanSelen) [inline src/dashboard.py:291]: 👍

**Review by DaanSelen — APPROVED** [review]
Looks good, small comment.

File: src/dashboard.py (hunk 3 above)
Comment (donaldzou) [reply src/dashboard.py:119]: To be honest I don't remember I put the `try` over there... but it doesn't seem to do anything to I just removed it haha

Comment (DaanSelen) [reply src/dashboard.py:119]: Yeah good for my review.

**Pull request merged**